From 9d938ac4ef5c10625840629f00f4a345f12cfd41 Mon Sep 17 00:00:00 2001
Message-ID: <9d938ac4ef5c10625840629f00f4a345f12cfd41.1705854493.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705854493.git.sharlatanus@gmail.com>
References: <cover.1705854493.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 20 Jan 2024 15:50:48 +0000
Subject: [PATCH 18/19] gnu: stackistry: Use G-Expressions.

* gnu/packages/astronomy.scm (stackistry): Update package style by using
G-Expressions. Apply indention.
[arguments]<#:tests?>: Use it instead of deleting check phase.
<#:phases>: Remove trailing #f from each lambda.

Change-Id: I30f3906719713f0d47dc2049564bc2691ba3cd5d
---
 gnu/packages/astronomy.scm | 79 +++++++++++++++++++-------------------
 1 file changed, 39 insertions(+), 40 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index c049e6e67f..99b22fa39e 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -1227,52 +1227,51 @@ (define-public stackistry
         (base32 "0rz29v33n0x0k40hv3v79ym5ylch1v0pbph4i21809gz2al5p7dq"))))
     (build-system gnu-build-system)
     (arguments
-     `(#:make-flags
-       (list
-        (string-append
-         "SKRY_INCLUDE_PATH=" (assoc-ref %build-inputs "libskry") "/include")
-        (string-append
-         "SKRY_LIB_PATH=-L" (assoc-ref %build-inputs "libskry") "/lib")
-        (string-append
-         "LIBAV_INCLUDE_PATH=" (assoc-ref %build-inputs "ffmpeg") "/include"))
-       #:phases
-       (modify-phases %standard-phases
-         ;; no configure and tests are provided
-         (delete 'configure)
-         (delete 'check)
-         (add-after 'unpack 'fix-paths
-           (lambda* (#:key outputs #:allow-other-keys)
-             (substitute* "src/main.cpp"
-               (("\"\\.\\.\", \"lang\"")
-                "\"../share/stackistry\", \"lang\""))
-             (substitute* "src/utils.cpp"
-               (("\"\\.\\.\", \"icons\"")
-                "\"../share/stackistry\", \"icons\""))
-             #t))
-         (replace 'install
-           ;; The Makefile lacks an ‘install’ target.
-           (lambda* (#:key outputs #:allow-other-keys)
-             (let* ((out (assoc-ref outputs "out"))
-                    (bin (string-append out "/bin"))
-                    (icons (string-append out "/share/stackistry/icons"))
-                    (lang (string-append out "/share/stackistry/lang")))
-               (copy-recursively "bin" bin)
-               (copy-recursively "icons" icons)
-               (copy-recursively "lang" lang))
-             #t)))))
+     (list
+      #:tests? #f ; No test target
+      #:make-flags
+      #~(list
+         (string-append
+          "SKRY_INCLUDE_PATH=" #$(this-package-input "libskry") "/include")
+         (string-append
+          "SKRY_LIB_PATH=-L" #$(this-package-input "libskry") "/lib")
+         (string-append
+          "LIBAV_INCLUDE_PATH=" #$(this-package-input "ffmpeg") "/include"))
+      #:phases
+      #~(modify-phases %standard-phases
+          ;; no configure and tests are provided
+          (delete 'configure)
+          (add-after 'unpack 'fix-paths
+            (lambda _
+              (substitute* "src/main.cpp"
+                (("\"\\.\\.\", \"lang\"")
+                 "\"../share/stackistry\", \"lang\""))
+              (substitute* "src/utils.cpp"
+                (("\"\\.\\.\", \"icons\"")
+                 "\"../share/stackistry\", \"icons\""))))
+          (replace 'install
+            ;; The Makefile lacks an ‘install’ target.
+            (lambda _
+              (let* ((out #$output)
+                     (bin (string-append out "/bin"))
+                     (icons (string-append out "/share/stackistry/icons"))
+                     (lang (string-append out "/share/stackistry/lang")))
+                (copy-recursively "bin" bin)
+                (copy-recursively "icons" icons)
+                (copy-recursively "lang" lang)))))))
     (native-inputs
      (list pkg-config))
-     (inputs
-      (list gtkmm-3 libskry ffmpeg-4))
-     (home-page "https://github.com/GreatAttractor/stackistry")
-     (synopsis "Astronomical lucky imaging/image stacking tool")
-     (description
-      "Stackistry implements the lucky imaging principle of astronomical
+    (inputs
+     (list gtkmm-3 libskry ffmpeg-4))
+    (home-page "https://github.com/GreatAttractor/stackistry")
+    (synopsis "Astronomical lucky imaging/image stacking tool")
+    (description
+     "Stackistry implements the lucky imaging principle of astronomical
 imaging: creating a high-quality still image out of a series of many (possibly
 thousands) low quality ones (blurred, deformed, noisy).  The resulting image
 stack typically requires post-processing, including sharpening (e.g. via
 deconvolution).  Such post-processing is not performed by Stackistry.")
-     (license license:gpl3+)))
+    (license license:gpl3+)))
 
 (define-public stellarium
   (package
-- 
2.41.0

